Applications/FileIO/SWMM/SWMMInterface.cpp

```diff
@@ -0,0 +1,1248 @@
+                sc.rain_gauge = i;
+
+        if (sc.rain_gauge == std::numeric_limits<std::size_t>::max())
+        {
+            ERR ("Rain gauge for subcatchment \"%s\" not found.", split_str[0].c_str());
+            return false;
+        }
+
+        sc.outlet =  std::numeric_limits<std::size_t>::max();
+        auto const it = name_id_map.find(split_str[2]);
+        if (it == name_id_map.end())
+        {
+            ERR ("Outlet node for subcatchment \"%s\" not found.", split_str[0].c_str());
+            return false;
+        }
+        sc.outlet = it->second;
+        sc.area = BaseLib::str2number<double>(split_str[3]);
+        _subcatchments.push_back(sc);
+        getline(in, line);
+    }
+
+    return true;
+}
+
+bool SwmmInterface::convertSwmmInputToLineMesh()
+{
+    if (_mesh != nullptr)
+    {
+        ERR ("Mesh already exists.");
+        return false;
+    }
+
+    std::string const inp_file_name (_base_name + ".inp");
+    if (!isSwmmInputFile(inp_file_name))
+        return false;
+
+    std::ifstream in ( inp_file_name.c_str() );
+    _id_nodename_map.clear();
+    std::vector< MeshLib::Node* > nodes;
+    std::string line;
+    while ( getline(in, line) )
+    {
+        if (line == "[COORDINATES]")
+        {
+            INFO ("Reading coordinates...");
+            if (!readCoordinates<MeshLib::Node>(in, nodes, _id_nodename_map))
+                return false;
+        }
+        /* todo: check if needed
+        if (line == "[VERTICES]")
+        {
+            INFO ("Reading vertices...");
+            if (!readCoordinates(in, nodes, _id_nodename_map))
+                return false;
+        }
+        */
+        if (line == "[SYMBOLS]")
+        {
+            INFO ("Reading symbols...");
+            std::vector<GeoLib::Point*> points;
+            std::vector<std::string> names;
+            if (!readCoordinates(in,points, names))
+                return false;
+            for (std::size_t i=0; i<points.size(); ++i)
+            {
+                GeoLib::Station stn (points[i], names[i]);
+                _rain_gauges.push_back(std::pair<GeoLib::Station, std::string>(stn, ""));
+            }
+        }
+    }
+
+    if (nodes.empty())
+        return false;
+
+    // After end of file is reached, create name-id-map and
+    // start reading again to get line elements and node data.
+    std::map< std::string, std::size_t> name_id_map;
+    std::size_t const n_nodes (nodes.size());
+    for (std::size_t i=0; i<n_nodes; ++i)
+        name_id_map[_id_nodename_map[i]] = i;
+    in.clear();
+    in.seekg(0, in.beg);
+
+    std::vector< MeshLib::Element* > elements;
+    std::vector<double> max_depth;
+    max_depth.resize(n_nodes);
+    std::size_t const n_types = 3;
+    std::array< std::size_t, n_types> n_elem_types;
+    while ( getline(in, line) )
+    {
+        if (line == "[RAINGAGES]")
+        {
+            if (!_rain_gauges.empty())
+                addRainGaugeTimeSeriesLocations(in);
```

Comment:
No idea. You could probably run simulations without rain if you'd just add BCs somewhere in the channel network. I'm no expert, though.